go/cmd/dolt/commands/log.go

```diff
@@ -110,576 +88,188 @@ func (cmd LogCmd) ArgParser() *argparser.ArgParser {
-	switch decorateOption {
-	case "short", "full", "auto", "no":
-	default:
-		return nil, fmt.Errorf("fatal: invalid --decorate option: %s", decorateOption)
+		handleErrAndExit(err)
 	}
-
-	opts := &logOpts{
-		numLines:    apr.GetIntOrDefault(cli.NumberFlag, -1),
-		showParents: apr.Contains(cli.ParentsFlag),
-		minParents:  minParents,
-		oneLine:     apr.Contains(cli.OneLineFlag),
-		decoration:  decorateOption,
+	if closeFunc != nil {
+		defer closeFunc()
 	}
 
-	err := opts.parseRefsAndTable(ctx, apr, dEnv)
+	query, err := constructInterpolatedDoltLogQuery(apr)
 	if err != nil {
-		return nil, err
+		return handleErrAndExit(err)
 	}
-
-	excludingRefs, ok := apr.GetValueList(cli.NotFlag)
-	if ok {
-		if len(opts.excludingCommitSpecs) > 0 {
-			return nil, fmt.Errorf("cannot use --not argument with two dots or ref with ^")
-		}
-		if len(opts.tableName) > 0 {
-			return nil, fmt.Errorf("cannot use --not argument with table")
-		}
-		for _, excludingRef := range excludingRefs {
-			notCs, err := doltdb.NewCommitSpec(excludingRef)
-			if err != nil {
-				return nil, fmt.Errorf("invalid commit %s\n", excludingRef)
-			}
-
-			opts.excludingCommitSpecs = append(opts.excludingCommitSpecs, notCs)
-		}
+	logRows, err := GetRowsForSql(queryist, sqlCtx, query)
+	if err != nil {
+		return handleErrAndExit(err)
 	}
 
-	return opts, nil
+	return logCommits(apr, logRows, queryist, sqlCtx)
 }
 
-func (opts *logOpts) parseRefsAndTable(ctx context.Context, apr *argparser.ArgParseResults, dEnv *env.DoltEnv) error {
-	// `dolt log`
-	if apr.NArg() == 0 {
-		return nil
-	}
+// constructInterpolatedDoltLogQuery generates the sql query necessary to call the DOLT_LOG() function.
+// Also interpolates this query to prevent sql injection.
+func constructInterpolatedDoltLogQuery(apr *argparser.ArgParseResults) (string, error) {
+	var params []interface{}
 
-	if strings.Contains(apr.Arg(0), "..") {
-		if apr.NArg() > 1 {
-			return fmt.Errorf("Cannot use two or three dot syntax when 2 or more arguments provided")
-		}
-
-		// `dolt log <ref>...<ref>`
-		if strings.Contains(apr.Arg(0), "...") {
-			refs := strings.Split(apr.Arg(0), "...")
-
-			for _, ref := range refs {
-				cs, err := getCommitSpec(ref)
-				if err != nil {
-					return err
-				}
-				opts.commitSpecs = append(opts.commitSpecs, cs)
-			}
+	var buffer bytes.Buffer
+	var first bool
+	first = true
 
-			mergeBase, verr := getMergeBaseFromStrings(ctx, dEnv, refs[0], refs[1])
-			if verr != nil {
-				return verr
-			}
-			notCs, err := getCommitSpec(mergeBase)
-			if err != nil {
-				return err
-			}
-			opts.excludingCommitSpecs = append(opts.excludingCommitSpecs, notCs)
+	buffer.WriteString("select commit_hash from dolt_log(")
 
-			return nil
+	writeToBuffer := func(s string, param bool) {
+		if !first {
+			buffer.WriteString(", ")
 		}
-
-		// `dolt log <ref>..<ref>`
-		refs := strings.Split(apr.Arg(0), "..")
-		notCs, err := getCommitSpec(refs[0])
-		if err != nil {
-			return err
+		if !param {
```

Comment:
I think we can remove the `param` param if we just wrap `s` in quotes when it's not a parameter, like

```
writeToBuffer("\"--table\"")
```